### ISSUE TITLE
DSL Extensions for Training

### ISSUE THREAD

**Issue by michalharakal** [sk-ai-net/skainet#39]
- [ ] Extend network DSL for training configuration
  - [ ] Add loss function specification
  - [ ] Implement optimizer configuration
  - [ ] Add training hyperparameters (learning rate, batch size, etc.)
  - [ ] Create validation configuration

- [ ] Implement training DSL
  - [ ] Create train block for defining training process
  - [ ] Add validation block for validation configuration
  - [ ] Implement callbacks system for training events
  - [ ] Add logging and visualization configuration
